README.md

```diff
@@ -1,35 +1,93 @@
+<hr/>
 
-```
-module.exports = {
-  development: {
-    store: {
-      type: "github",
-    },
+This plugin deploy your app to a specified repository. It's to be used to deploy your application in github pages. To see this plguin in action take a look in  [ember-cli-deploy-github-example](https://github.com/dukex/ember-cli-deploy-github-example) repository.
+
+## What is an ember-cli-deploy plugin?
+
+A plugin is an addon that can be executed as a part of the ember-cli-deploy pipeline. A plugin will implement one or more of the ember-cli-deploy's pipeline hooks.
+
+For more information on what plugins are and how they work, please refer to the [Plugin Documentation](http://ember-cli.github.io/ember-cli-deploy/plugins).
+
+## Quick Start
+
+To get up and running quickly, do the following:
+
+* Ensure [ember-cli-deploy-build](https://github.com/zapnito/ember-cli-deploy-build) is installed and configured.
+
+* Install this plugin
 
-    assets: {
-      type: "github",
-      branch: "master", // default is gh-pages
-      repository: "git@github.com:user/repo.git",
-    }
-  }
-}
+```
+$ ember install ember-cli-deploy-github
 ```
 
-## note about push
+* Place the following configuration into config/deploy.js
 
-The current version will always make with push with force(```--force```)
+```
+ENV.github = {
+  repository: '<your-repository>'
+};
+```
 
+* Run pipeline
+
+```
+$ ember deploy
+```
 
 ## Installation
 
-* `git clone` this repository
-* `npm install`
+```
+$ ember install ember-cli-deploy-github
+```
+
+## ember-cli-deploy Hooks Implemented
+
+For detailed information on what plugin hooks are and how they work, please refer to the [Plugin Documentation](http://ember-cli.github.io/ember-cli-deploy/plugins).
+
+* ```configure```
+* ```upload```
+
+## Configuration Options
+
+For detailed information on how configuration of plugins works, please refer to the [Plugin Documentation](http://ember-cli.github.io/ember-cli-deploy/plugins).
+
+### branch
+
+The target branch on remote repository where the plugin will push app. By default the plugin uses ```gh-pages```, for more information about branch please refer to [Github Pages](https://pages.github.com/)
+
+Default: ```gh-pages```
+
+### repository
+
+The repository to which the plugin will push your app.
+
+TODO: add examples
+
+Default: ```undefined```
+
+### distDir
+
+The root directory where your app was built. By default, this option will use the distDir property of the deployment context, provided by [ember-cli-deploy-build](https://github.com/zapnito/ember-cli-deploy-build).
+
+Default: ```context.distDir```
+
+## Prerequisites
+
+The following properties are expected to be present on the deployment context object:
+
+* ```distDir``` (provided by [ember-cli-deploy-build](https://github.com/zapnito/ember-cli-deploy-build)).
+
+## Plugin Warnings
+
+The current version will always deploy using a force push(i.e. ```git push --force```),
```

Comment:
No, it's easy make this configurable. But the plugin need do work, for example fetch/rebase, maybe handle conflict, etc. I'll try to implement this today night.